File: codes/ruby/chapter01/answer_A02.rb
# 入力
n, x = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)

# 全探索（変数 answer は「既に x が見つかったかどうか」を表す）
answer = false
a.each do |elem|
  answer = true if elem == x
end

# 出力
if answer
  puts "Yes"
else
  puts "No"
end

File: codes/elixir/chapter01/answer_A02.exs
ar = IO.read(:line) |> String.split() |> Enum.map(&String.to_integer/1)
x = String.trim(IO.gets("")) |> String.to_integer

answer = Enum.any?(ar, fn(i) -> i == x end)
IO.puts(answer)

if answer do
  IO.puts("Yes")
else
  IO.puts("No")
end
